community/contributing/pr_review_guidelines.rst

```diff
@@ -0,0 +1,396 @@
+<https://chat.godotengine.org>`_, in the appropriate channel or in #general.
+Other teams may already be tagged for review, so you can also wait or ask for
+their assistance.
+
+6. Approve the pull request
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After reviewing the code, if you think that the code is ready to be merged into
+the engine, then go ahead and "approve" it. Make sure to also comment and
+specify the nature of your review (i.e. say whether you ran the code locally,
+whether you reviewed for style as well as correctness etc.). Even if you are not
+an engine maintainer, approving a pull request signals to others that the code
+is good and likely solves the problem the PR says it does. Approving a pull
+request as a non-engine maintainer does not guarantee that the code will be
+merged, other people will still review it, so don't be shy.
+
+Code style review
+-----------------
+
+Generally speaking, we aim to conduct a code review before a style/clarity
+review as contributors typically want to know if their general approach is
+acceptable before putting in the effort to make nitpicky changes to style. In
+other words, maintainers shouldn't ask contributors to change the code style of
+code that may need to be rewritten in subsequent reviews. Similarly, maintainers
+should avoid asking for contributors to rebase PRs if the PR has not been
+reviewed.
+
+That being said, not everyone feels confident enough to provide a review on code
+correctness, in that case, providing comments on code style and clarity ahead of
+a more substantive code review is totally appropriate and more than welcome.
+
+In practice the code style review can be done as part of the substantive code
+review. The important thing is that both the substantive code and the code style
+need to be reviewed and considered before a pull request is merged.
+
+When reviewing code style pay particular attention to ensuring that the pull
+request follows the :ref:`doc_code_style_guidelines`.
+
+While ``clang-format`` and various CI checks can catch a lot of inconsistencies,
+they are far from perfect and are unable to detect some issues.
+For example, check that:
+
+  * The style of header includes is respected.
+  * Identifiers use ``snake_case`` and follow our naming conventions.
+  * Method parameters start with ``p_*`` or ``r_*`` (if they are used to return
+    a value).
+  * Braces are used appropriately, even for one-liner conditionals.
+  * Code is properly spaced (exactly one empty line between methods, no
+    unnecessary empty lines inside of method bodies).
+
+.. note::
+
+    This list is not complete and doesn't aim to be complete. Refer to
+    the linked style guide document for a complete set of rules. Keep
+    in mind that ``clang-format`` may not catch things you hope it would,
+    so pay attention and try to build a sense of what exactly it can and
+    cannot detect.
+
+Merging pull requests
+---------------------
+
```

Comment:
Can't help but notice that we don't have any mention for the fact that you shouldn't merge your own pull requests. I think it's important to spell out, even though we repeat a few times that merging is done by the production team or team leaders (a team leader shouldn't merge their own PRs either, normally).

Reply:
Its mentioned under ``Get feedback from the right people/teams``. I went with a pretty soft approach on this as right now many PRs are merged and reviewed by Akien alone. We can strengthen the language once we formalize our practice a bit more. As it is right now we would open ourselves to criticism if we use strong language

Reply:
That section talks about not merging PRs based on your own sole review. I'm talking about merging a PR that you yourself have authored. It was point 0 in one of the sections of my original PR, if that helps.

Even Akien doesn't like to merge his own PRs, but there is often little help with some stuff that he changes. It would be nice to underline that self-merges are not expected nor desired. Even if to show intent.